src/core/quantum_circuit.jl

```diff
@@ -154,6 +154,35 @@ function compare_circuits(c0::QuantumCircuit,c1::QuantumCircuit)::Bool
     return compare_kets(ψ_0,ψ_1)
 end
 
+"""
+    Base.in(gate_type::Type{<:AbstractGate}, circuit::QuantumCircuit)::Bool
```

Comment:
The usual Base.in() assert that an instance is present, not a type. this seems to break cannon. Maybe call it has_gate_type() instead?

Reply:
I've renamed the function to `circuit_contains_gate_type`